### PR TITLE
Task 34/Add Appbar Sidenav

### PR DESCRIPTION
**Pull request by tjosepo**
# Description

This adds the code of the sidenav and appbar components.

Issue reference: #34 

## Type of change

Please delete options that are not relevant.

- [ ] BUG (Fixing or documenting a bug)
- [ ] FEAT (Add a new feature to the project)
- [ ] CHORE (Something that needs to be done outside of the issue)
- [ ] DOCS (Implement documentation)
- [X] TASK (A task that must be done to complete an issue)

# Checklist:

- [X] My pull request references the issue in the comments
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/partials/appbar/Appbar.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import Appbar from './Appbar';
+
+let showSidenav: boolean;
+const toggleSidenav: any = (val: boolean) => showSidenav = val;
+
+test('toggles sidenav state from off to on', () => {
+  showSidenav = false;
+
+  render(<Appbar {...{showSidenav, toggleSidenav}}/>);
+  screen.getByRole("button").click()
```
Comment (phong1233) [inline frontend/src/components/partials/appbar/Appbar.test.tsx]:
```suggestion
  screen.getByRole("button").click();
```

File: frontend/src/components/partials/appbar/Appbar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  expect(showSidenav).toBeTruthy();
+});
+
+test('toggles sidenav state from on to off', () => {
+  showSidenav = true;
+
+  render(<Appbar {...{showSidenav, toggleSidenav}}/>);
+  screen.getByRole("button").click()
```
Comment (phong1233) [inline frontend/src/components/partials/appbar/Appbar.test.tsx]:
```suggestion
  screen.getByRole("button").click();
```

**Review by phong1233 — APPROVED** [review]
Magnifique!

File: frontend/src/components/partials/appbar/Appbar.test.tsx (hunk 2 above)
Comment (tjosepo) [reply frontend/src/components/partials/appbar/Appbar.test.tsx]: nice

**Review by sebastien-blain — APPROVED** [review]
Well done Mr. Josepovic.

**Pull request merged**